backend/tests/unit/controllers/posts-controller.test.js

```diff
@@ -245,16 +195,16 @@ describe('getPostByCategoryHandler', () => {
   });
 
   it('should handle errors - Invalid Category', async () => {
-    const req = { params: { category: 'Invalid Category' } };
+    const req = createRequestObject({ params: { category: 'Invalid Category' } });
 
     await getPostByCategoryHandler(req, res);
 
     expect(res.status).toHaveBeenCalledWith(400);
     expect(res.json).toHaveBeenCalledWith({ message: 'Invalid category' });
   });
 
-  it('should handle errors', async () => {
-    const req = { params: { category: 'Travel' } };
+  it('should handle errors - Internal Server Error', async () => {
+    const req = createRequestObject({ params: { category: validCategories[1] } });
```

Comment:
Please check everywhere, the change is needed.

Pro tip: If it soo happens that you cannot describe the test cases properly even after thinking, then we can delete that test case :)